java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/EmailSenderService.java

```diff
@@ -1,285 +1,95 @@
-		contentStr.append("<br><br>");
-		contentStr.append("New device has been connected to OsmAnd Cloud. Your activation code is <b>"+token+"</b><br>");
-		contentStr.append("You can also use <a clicktracking=\"off\" href='https://osmand.net/premium/device-registration?token=" + token
-				+ "'>following link<a> to open it with OsmAnd.");
-		contentStr.append("<br><br>");
-		contentStr.append("Best Regards, <br>OsmAnd Team");
-
-		Content content = new Content("text/html", contentStr.toString());
-		Mail mail = new Mail(from, topic, to, content);
-		mail.from = from;
-		try {
-			Response response = wrapper.send(mail);
-			LOGGER.info("Response code: " + response.getStatusCode());
-		} catch (Exception e) {
-			LOGGER.warn(e.getMessage(), e);
-		}
+    public void sendOsmAndCloudRegistrationEmail(String email, String token, boolean newUser) throws FileNotFoundException, UnsupportedEncodingException {
+		String subject = newUser ? "Welcome to OsmAnd Cloud" : "Register new device to OsmAnd Cloud";
+	    boolean ok = new EmailSenderTemplate()
+			    .load("cloud/register")
+			    .set("SUBJECT", subject)
+			    .set("TOKEN", token)
+			    .to(email)
+			    .send()
+			    .isSuccess();
+	    LOGGER.info("sendOsmAndCloudRegistrationEmail to: " + email + " (" + ok + ")");
 	}
     
-    // 6503d80a-4574-461d-95f1-2bd0c43392b8
-    public boolean sendPromocodesEmails(String mailTo, String templateId, String promocodes) {
-    	LOGGER.info("Sending mail to: " + mailTo);
-        Email from = new Email(DEFAULT_MAIL_FROM);
-        from.setName("OsmAnd");
-        Mail mail = new Mail();
-        mail.from = from;
-        Personalization personalization = new Personalization();
-        for(String mt : mailTo.split(",")) {
-        	if(!Algorithms.isEmpty(mt)) {
-        		personalization.addTo(new Email(mt));
-        	}
-        }
-        mail.addPersonalization(personalization);
-        mail.setTemplateId(templateId);
-        MailSettings mailSettings = new MailSettings();
-        FooterSetting footerSetting = new FooterSetting();
-        footerSetting.setEnable(true);
-        String footer = String.format("<center style='margin:5px 0px 5px 0px'>"
-        		+ "<h2>Your promocode is '%s'.</h2></center>", promocodes) ;
-        footerSetting.setHtml("<html>"+footer+"</html>");
-        mailSettings.setFooterSetting(footerSetting);
-        mail.setMailSettings(mailSettings);
-        try {
-            Response response = wrapper.send(mail);
-            LOGGER.info("Response code: " + response.getStatusCode());
-            return true;
-        } catch (Exception e) {
-        	LOGGER.warn(e.getMessage(), e);
-        	return false;
-		}
+    public boolean sendPromocodesEmails(String mailTo, String templateId, String promocodes) throws FileNotFoundException, UnsupportedEncodingException {
+	    boolean ok = new EmailSenderTemplate()
+			    .load(templateId) // should be "promocode/ios" or "promocode/anroid"
```

Comment:
To https://osmand.net/admin/info#giveaway need add tips where to create new template 
( https://github.com/osmandapp/web-server-config/tree/main/templates/email )

Reply:
Added tips and link to edit templates (github repo)